### ISSUE TITLE
Ability to add custom fields & checkboxes

### ISSUE THREAD

**Issue by jared-gs** [akiko-pusu/redmine_issue_templates#39]
Great plugin, I've been using it a lot lately.

One addition to improve upon it would be the ability to add custom fields and fields from other plugins.  

The plugin I specifically had in mind is the Redmine checklist plugin (https://www.redmine.org/plugins/redmine_checklists).  I have several issue templates, and unfortunately I still need to add checklists manually.  Having the ability integrated these into Issue Templates would improve productivity.

**Comment by akiko-pusu** [comment]
Hi, @jared-gs 
Thanks for your request.
It seems checklist plugin PRO version has now template feature. (Unfortunately, Free version is not supported, so that I have not tried this plugin yet...)
Please let me confirm if you use checklist PRO version or not? (Using Free version?)

**Comment by jared-gs** [comment]
I haven't used the PRO version.  I'm using the Free version.  The ability to add "3rd-party" fields (checkboxes, custom fields, etc) to [redmine_issue_templates](https://github.com/akiko-pusu/redmine_issue_templates/) would be a great feature to have, if it were possible.

**Comment by akiko-pusu** [comment]
Hi, @jared-gs 
Thanks for your additional information.
I’m trying to implement on feature/work-with-checklists branch.

My planning is following:

1. Register checklist as one of template
- Add field to store checklist contents on IssueTemplate table.
 - Save checklist value as json.

2. Load checklist when creating new issue
- Load selected template with checklist json object.
- Add checklist form and click “+” (plus) button via JavaScript after loading template data.

So, I’m trying to input and save checklists via JavaScript. 
I don’t want to modify Checklist plugin itself.
If you have not installed Checklist plugin, this action is ignored.

Could you please this branch try if you have time to share?
Or you can try and confirm its working at my demo site.

Please visit http://redmine-akiko.ruby.iijgio.com/
(Login information is here: http://redmine-akiko.ruby.iijgio.com/projects/sample-project/wiki)

Now, register checklist at template create/edit form is only for project template only.
If this approach is acceptable, I’m going to update global template too.

**Comment by akiko-pusu** [comment]
Hi, @jared-gs 
Thanks for your additional information.
I’m trying to implement on feature/work-with-checklists branch.

My planning is following:

1. Register checklist as one of template
- Add field to store checklist contents on IssueTemplate table.
 - Save checklist value as json.

2. Load checklist when creating new issue
- Load selected template with checklist json object.
- Add checklist form and click “+” (plus) button via JavaScript after loading template data.

So, I’m trying to input and save checklists via JavaScript. 
I don’t want to modify Checklist plugin itself.
If you have not installed Checklist plugin, this action is ignored.

Could you please this branch try if you have time to share?
Or you can try and confirm its working at my demo site.

Please visit http://redmine-akiko.ruby.iijgio.com/
(Login information is here: http://redmine-akiko.ruby.iijgio.com/projects/sample-project/wiki, 
http://redmine-akiko.ruby.iijgio.com/projects/sample-project/wiki#Account-info)

Now, register checklist at template create/edit form is only for project template only.
If this approach is acceptable, I’m going to update global template too.

**Comment by akiko-pusu** [comment]
Please note: 
If you try feature/work-with-checklists branch, migration is required. (Because one column to store checklist value is generated.)

Any suggestion, question will be welcomed.
Also, pull request will be appreciated!

**Comment by jared-gs** [comment]
Akiko,

You're so awesome.  Thank you for doing this.

**Comment by akiko-pusu** [comment]
@jared-gs Thanks for your replying.
Though above branch is quite rough, l would continue to implement if you accept such approach, via JavaScript.
Not only checklist plugin, but also ather fields may be handled with the same approach.

**Comment by akiko-pusu** [comment]
Hi, feature to integrate with Checklist plugin was released in v0.1.3.
Other request such as builtin fields or custom fields.

**Comment by letic** [comment]
Great work Akiko,

Are you planning to also implement the ability to use custom fields ? 

Thanks in advance
Cheers

**Comment by akiko-pusu** [comment]
Hi, @letic  and all. 
Thanks for the kind comments above.
Sorry that other request such as builtin fields or custom fields are not implemented yet....

I'll do what I'm supposed to do. So please cheer for me :)
Any ideas to implement or related plugin's informations are also highly appreciated.

**Comment by todhunter** [comment]
Hi Akiko
It would be nice to add support of tagging in templates 
RedmineUp has great plugin for tagging (https://www.redmineup.com/pages/plugins/tags)
This is only thing that i really need to use yours plugin

**Comment by gessel** [comment]
Hi Akiko,

I saw that my report was a duplicate of this ongoing issue.  

I created templates both as global and as a project.  Both had the same number of checklist items set up like this:

![image](https://user-images.githubusercontent.com/3670635/46576557-5836bf00-c9d5-11e8-8b8b-37c47c5c9a52.png)

When I create a new template of the matching tracker, it looks good:

![image](https://user-images.githubusercontent.com/3670635/46576564-7b616e80-c9d5-11e8-9d9a-f76317e5833f.png)

But if I change the status from "new" to "in progress" - or save the ticket and check it again, the checklist is truncated 

![image](https://user-images.githubusercontent.com/3670635/46576575-a1870e80-c9d5-11e8-9ed5-87fe0c4d28c9.png)

After creating, some subset of the checklist items are visible (this time 4 - I checked all but two on creation).

![image](https://user-images.githubusercontent.com/3670635/46576839-a9e34780-c9dd-11e8-83de-6911193712f7.png)

Would be super cool if it worked!



```
Environment:
  Redmine version                3.4.6.stable
  Ruby version                   2.4.4-p296 (2018-03-28) [amd64-freebsd10]
  Rails version                  4.2.10
  Environment                    production
  Database adapter               Mysql2
SCM:
  Filesystem                     
Redmine plugins:
  periodictask                   3.3.0
  redmine_checklists             3.1.11 (free)
  redmine_custom_reports         0.1.5
  redmine_dmsf                   1.6.1
  redmine_email_images           0.1.1
  redmine_highlightjs            1.0.3
  redmine_image_clipboard_paste  3.3.0
  redmine_issue_templates        0.2.0
  redmine_sortable_table         0.99.1
  redmine_tweaks                 1.0.2
```

**Comment by gessel** [comment]
My error seems to be only with waterfox, chrome seems to work fine.

**Comment by slydlake** [comment]
Hey there!

Is there a plan to fill out custom fields in the future?

**Comment by akiko-pusu** [comment]
@slydlake @letic 
Hi, I've just merged above PR into my develop branch.
Since this is an experimental feature, please be careful if you hope to try it. 
Migration is required and some browsers may not work fine.

Hope this would be any help.
And if you still have interests in this feature, feedback would be highly appreciated!

**Comment by akiko-pusu** [comment]
merged into master.
please re-open or report new issue!

**Comment by akiko-pusu** [comment]
@jared-gs @slydlake @letic 
Hi, I've released v1.0.0 to support custom fields and builtin-fields.
Since this is an experimental feature, please be careful if you hope to try it.

Please see https://github.com/akiko-pusu/redmine_issue_templates/releases/tag/1.0.0
I'm looking forward to your feedback.
